[Bug]: Report Generation Failure for ballerinax/postgresql with the New Testerina Framework

### Description

With the new testerina framework, the postgres module test report generation fails with the following output.

```
               [fail] testInOutParameterArray:

                    Timestamp timezone range array does not match.
                         
                        expected: <postgresql:TimestamptzRange[]> '[{"upper":"2010-01-01 14:00:00+00","lower":"2010-01-01 12:00:00+00","upperboundI
                        nclusive":true,"lowerboundInclusive":false}]'
                        actual  : <(postgresql:TimestamptzRange & readonly)[] & readonly> '[{"upper":"2010-01-01 19:30:00+05:30","lower":"2010-01-01 17:30:00+05:30","upper
                        boundInclusive":true,"lowerboundInclusive":false}]'
                        


                382 passing
                1 failing
                0 skipped
ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in
https://github.com/ballerina-platform/ballerina-lang/issues with the
log you get below and your sample code.

We thank you for helping make us better.

[2022-11-01 10:17:35,087] SEVERE {b7a.log.crash} - com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 21818 path $.tests[382].failureMessage 
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 21818 path $.tests[382].failureMessage
        at com.google.gson.Gson.fromJson(Gson.java:978)
        at com.google.gson.Gson.fromJson(Gson.java:901)
        at io.ballerina.cli.utils.TestUtils.loadModuleStatusFromFile(TestUtils.java:222)
        at io.ballerina.cli.task.RunTestsTask.execute(RunTestsTask.java:219)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.TestCommand.execute(TestCommand.java:303)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
Caused by: com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 21818 path $.tests[382].failureMessage
        at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1562)
        at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:490)
        at com.google.gson.stream.JsonReader.hasNext(JsonReader.java:413)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:216)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at com.google.gson.Gson.fromJson(Gson.java:963)
        ... 7 more
```

### Steps to Reproduce

- Build the ballerina-lang repo and publish to the maven local repo.
- Use that lang version to build the ballerinax/postgresql module with tests.

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Test Framework

### Related issue(s) (optional)

https://github.com/ballerina-platform/ballerina-lang/issues/38092

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

--- a/misc/testerina/modules/testerina-core/src/main/ballerina/report.bal
+++ b/misc/testerina/modules/testerina-core/src/main/ballerina/report.bal
@@ -147,7 +147,7 @@ function moduleStatusReport(ReportData data) {
     data.failedCases().forEach(result => tests.push({
         "name": result.fullName(),
         "status": "FAILURE",
-        "failureMessage": result.message()
+        "failureMessage": replaceDoubleQuotes(result.message())
     }));
     data.skippedCases().forEach(result => tests.push({
         "name": result.fullName(),
@@ -167,4 +167,16 @@ function moduleStatusReport(ReportData data) {
     if err is error {
         println(err.message());
     }
+}
+
+function replaceDoubleQuotes(string originalString) returns string {
+    string updatedString = "";
+    foreach string chr in originalString {
+        if chr == "\"" {
+            updatedString += "\\\"";
+        } else {
+            updatedString += chr;
+        }
+    }
+    return updatedString;
 }

